### PR TITLE
Mutators tests updated

### PR DESCRIPTION
**Pull request by JunyungKim**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

1. 
There are a couple of updates in Mutators.py of GA. 
a. Bugs coming from inversionMutators are fixed. 
b. Codes are updated for the case that users placed mutation locations inversely (e.g., [3,1]  rather than [1,3])

2.  Unit tests for all Mutators are developed. 
3. A unit test for unifromCrossover is developed.
4. "tests" file in projects/raven/tests/framework/unit_tests/Optimizers  is updated.

##### What are the significant changes in functionality due to this change request?
1. User can use inversionMutation of Generic Algorithm. Previously, there was an error out.
2. All mutators are tested. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [ ] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: ravenframework/Optimizers/crossOverOperators/crossovers.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def uniformCrossover(parents,**kwargs):
     children1,children2 = uniformCrossoverMethod(parent1,parent2,crossoverProb)
     children[index]   = children1
     children[index+1] = children2
-    index = index + 1
+    index +=  2
```
Comment (mandd) [inline ravenframework/Optimizers/crossOverOperators/crossovers.py:106]: why index+= 2 instead of index+= 1?

Comment (JunyungKim) [reply ravenframework/Optimizers/crossOverOperators/crossovers.py:106]: image

if index += 1, then one of the children values are overwritten as iteration goes.

Comment (JunyungKim) [reply ravenframework/Optimizers/crossOverOperators/crossovers.py:106]: The figure below is the result of uniformCrossover unit test if index += 1:
image

Comment (JunyungKim) [reply ravenframework/Optimizers/crossOverOperators/crossovers.py:106]: The figure below is the result if index += 2:
image

**Review by mandd — APPROVED** [review]

**Review by mandd — APPROVED** [review]

**Pull request merged**